data-mermaid/mermaid-webapp
# feat: 454-related. This isnt a bug 'fix' likely, but it when a user toggles back online after being offline and the internet or server is unresponsive, the app will find out much faster and it will turn itself back offline.

**Description:** There is a bug I cant reproduce (you can see some of my attempts in the ticket comments). Basically it was observed by Emily out in the real world with a crap internet connection that when returning online the user gets stuck on a loading screen. Emily tried to fix it by immediately turning the app offline, and it was still stuck. All I observed was a slow loader when turning the app back online (it waits for the health check to realize the internet is crapped out). to at least speed that up for UX reasons, not to fix the bug Emily saw:
- is to give axios a timeout for health checks, 
- and also to manually trigger a health check when the user toggles online (to they dont have to wait until the next scheduled one)

**Steps to test**
1. run app locally, point it to remote dev server (https://dev-api.datamermaid.org/v1)
2. set the app to offline mode manually
3. unscrew cable internet from router
4. turn the app online manually, it should be online for a moment before turning itself back offline. The app should load and operate like normal. 
5. you should not be able to turn the app online again until the next healthcheck determines that the internet/server is operating (after you plug your internet cable back into your modem)
